### PR TITLE
Add Kayle Blog

### PR DESCRIPTION
**Pull request by arsenstorm**
This pull request adds Kayle’s Blog.

@wanapichpongsa we need to discuss our first blog post, I’ve written it, but it doesn’t feel quite right.

### CODE REVIEW

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+Kayle started as a school project back in February of 2023; since then, we’ve experimented with different approaches to content moderation, 
+we’ve spoken to stakeholders of all sizes and iterated to reach the degree of understanding that a content moderation platform needs to have.
+
+## Our mission
+
+When we were children, the Internet exposed us to material that is still too difficult to describe and that we should never have seen.
+
+As time goes on, we continue to observe some of the *worst in humanity*—things that no one should have to see.
+
+This has to stop.
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx]: It's good that you want to remain concise, but the buildup to "This has to stop" is in my opinion – incomplete.

Points we need to elaborate:
1. For such an important issue, there exists incomplete solutions. Enterprise grade only available to big-tech, leaving majority of platforms with mediocre solutions. (I know we cover it in the next section, but I think we should mention in mission -> expand in next section)
2. I want to explicitly state the need that content moderation should be 'the tool that is available everywhere'

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+So we’re building Kayle to prevent harmful content from being easily accessible and to protect everyone, **especially children**.
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:33]: That's why sounds better than 'So'

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+## What we’ve learnt
+
+In the past year and a half, we’ve learned that several factors can make or break a content moderation service.
+
+The biggest factor? **Cost**.
+
+From speaking with those who need moderation, we continue to see that the most significant reason they opt out of using **any** form of moderation is the cost of implementing it.
+
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:44]: It seems a bit weird that the biggest factor (cost) visually has a smaller section than UX.

Comment (wanapichpongsa) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:44]: Analogy for thought: Moderation costs have become 'unforgiving'/significantly increased the average platform's operating expenses.

Platforms have to ration their costs to their most essential core expenses. No more moderation, bad actors set a horrible UX. Violation of Google Adsense = taken down

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+**A good content moderation service is <ins>cheap</ins>—even at much higher usage levels.**
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:45]: I think this is more of a 'political statement'

Great content moderation **should** be affordable – especially at higher usage levels/ larger scale.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+A commonly overlooked factor is **user experience**, which refers to several different types of users.
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:49]: 'different types of users:'

Comment (wanapichpongsa) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:49]: I would be specific to say 'Furthermore, a quality that existing services dangerously overlooked is the**user experience**, affecting the most important kind of users to a web platform: <= use colon

 (if you want to use app, then use web-app)

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+3. The business leaders
+
+Here’s why:
+1. A platform’s users don’t want poor moderation to affect them, whether reports go unaddressed or their content is inaccurately removed.
+2. The moderators want an easy-to-use platform to manage reports, moderate content, or suspend users—they **need** the platform to be clear from day one, not after weeks of use.
+3. The business leaders need a service that can give them peace of mind, knowing their platform is safe.
+
+Another commonly overlooked factor is **developer experience**.
+
+Excellent developer experience (DX) is crucial for any software-based company, and a content moderation platform is no exception.
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:65]: I think its worth mentioning how most of these tools (GCP), (AWS) aren't specific to content moderation, and are so complicated that there are 4-week long certificate programmes so you can gain literacy in them. 

Have sympathy for the developers! (after all, they're the ones who make sure we get the best Saas (use general term if you want) products. Poor DX = Paying OT/Raises for upskiling to deal with crap technology.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+## What we’re doing
+
+We’re building Kayle to be the **greatest** content moderation platform in the world.
+
+And we’re building Kayle in public with our entire codebase open-source for several reasons:
+- Anyone should be able to contribute to Kayle.
+- Everyone needing content moderation should be able to access the service without using our cloud platform (i.e., anyone should be able to self-host Kayle).
+
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:76]: In fact, we need global contribution (too many contextual nuances to count), so we're counting on you benevolent devs to help us! => win arms race against content generation that can be used for horrible things.

^ My thought process, obviously don't quote by verbatim.

**Review by wanapichpongsa — CHANGES_REQUESTED** [review]
We got the essential points covered. I gave some comments to help the article make more sense.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 1 above)
Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx]: I disagree—it acts as concise transition between problem and solution,

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 2 above)
Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:33]: Agreed

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 4 above)
Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:45]: Agreed

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 3 above)
Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:44]: I agree; it seems a bit weird.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 5 above)
Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:49]: I disagree regarding the web platform point.

I’ve changed it to: `A commonly overlooked factor is **user experience**, which refers to the experience of several different types of users.`

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 6 above)
Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:65]: I agree, although I'm unsure how to phrase it.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 1 above)
Comment (wanapichpongsa) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx]: Right, but I still think the characterization of the problem is incomplete. This problem has been ongoing and addressed for over a decade now, along with the many solutions than came about. (From what I've read, its quite unclear how our mission is different from the 10s of solutions that came before us)

I strongly believe we should elaborate how **accessibility/availability** to content moderation is the crucial factor to why many parts of the internet is unsafe, and at risk of being even more dangerous.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 5 above)
Comment (wanapichpongsa) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx]: I think the transition from the previous sentence (concluding sentence for high cost) to the UI statement is not cohesive.

+ `which refers to the experience of several different types of users.` just sounds weird and redundant to me. 

`which refers to the experience` is implied by the term user experience, whether you're techincal or not. 

**I'd focus more on what exactly you mean by user experience (it tends to be complicated, and is difficult to use) => affecting a variety of users/affecting important users**

`several different types` -> `a variety`

`:` as you're listing 3 types straight afterwards.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+The biggest factor? **Cost**.
+
+From speaking with those who need moderation, we continue to see that the most significant reason they opt out of using **any** form of moderation is the cost of implementing it.
+
+**A good content moderation service is <ins>cheap</ins>—even at much higher usage levels.**
+
+A commonly overlooked factor is **user experience**, which refers to several different types of users.
+1. A platform’s users
+2. The moderators
+3. The business leaders
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:52]: `Their moderators`
`Their business leaders`

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 5 above)
Comment (wanapichpongsa) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx]: `poorly designed user experience` <= something like that sounds better to me

Comment (wanapichpongsa) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx]: Plus, if you mention `platform` in the sentence that preludes to your list of user types, you can list out:

1. Their users
2. Their moderators
3. Their business leaders

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:53]: I notice that there's a lot of 'optional' implications by continuous usage of the word 'want'. 

I suggest we phrase it with the whole, ugly truth – content moderation is necessary

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+Here’s why:
+1. A platform’s users don’t want poor moderation to affect them, whether reports go unaddressed or their content is inaccurately removed.
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:55]: 1. User-Generated-Content is the lifeblood to any online community. 

2. Poor moderation leads to disturbing content traumatizing users, or unjust removal of content. Both accuse the platform of not enforcing their content policies – content that built the culture that made their platform thrive in the first place. 

After writing more blogs <a href="https://www.blog.kayle.ai/why-twitter-lost-advertising-revenue>(see case studies)<a>

or `https://www.blog.kayle.ai/blogs/search?query=poor user experience`

Comment (wanapichpongsa) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:55]: I was going to say:

1. User-Generated-Content is the lifeblood to any online community. 

2. Poor moderation leads to disturbing content traumatizing users, or unjust removal of content. Both accuse the platform of either being too extreme, or anti-freedom of speech.

But we aren't here to set the policies. We're here to enforce them.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+2. The moderators want an easy-to-use platform to manage reports, moderate content, or suspend users—they **need** the platform to be clear from day one, not after weeks of use.
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:56]: This sentence needs to be more cohesive. 

`Most platforms take weeks to get used to. To avoid costly mistakes, moderators need a platform that can be used effectively as soon as they log in. No user manuals, no pricey consultation calls {that nobody wants to pay for.` 

remove '{..}' if unnecessary.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+3. The business leaders need a service that can give them peace of mind, knowing their platform is safe.
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx]: `need a reliable service that'll keep their platform safe, so they can focus on improving their product`

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 8 above)
Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:52]: Pretty much—when I was writing it, I couldn’t work out the best way of phrasing it.

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 9 above)
Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:53]: Maybe

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 1 above)
Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx]: I agree—let's discuss this on a call

File: platform/app/(blog)/blog/introducing-kayle/page.mdx (hunk 6 above)
Comment (wanapichpongsa) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:65]: `Another commonly ... **developer experience** (DX)

`Most software-based companies use cloud platforms such as Amazon Web Services or Google Cloud Platform. Many of our waitlist members have complained about how the complexity of cloud platforms forces their hand to hire developer teams that are larger than necessary.` (and watch their costs burst through the roof/ceiling)

`Rather than improving their product, waste time and resources into learning certificate courses that are weeks long.`

**Need to combine these two somehow...**
1. `We aim to make Kayle's content management platform so simple and straightforward, non-technical moderators will know how to use our platform without needing to be educated by video tutorials.`
2. APIs simple and customizable.

I referred to severe pain points received by 'waitlisters':

1. Hiring a larger team of engineers than needed because of cloud platform complexity.
3. Skill gap puts a platform at risk of content moderation **failure**. An active platform can't afford inactive content moderation for a few hours, let alone a day.
4. Cloud platforms aren't specifically made for content moderation, which is why features are limited.

**P.S.** We understand that for some platforms, cloud platforms are essential. We're talking about symbiosis between cloud platforms and Kayle, not removal of one completely. (i.e. it's just incomplete!)

File: platform/app/(blog)/blog/introducing-kayle/page.mdx [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+Another commonly overlooked factor is **developer experience**.
+
+Excellent developer experience (DX) is crucial for any software-based company, and a content moderation platform is no exception.
+
+## What we’re doing
+
+We’re building Kayle to be the **greatest** content moderation platform in the world.
```
Comment (wanapichpongsa) [inline platform/app/(blog)/blog/introducing-kayle/page.mdx:71]: I'd avoid self-proclaimed greatness.

I'd rephrase as `go-to`

Comment (arsenstorm) [reply platform/app/(blog)/blog/introducing-kayle/page.mdx:71]: fair enough

**Review by wanapichpongsa — APPROVED** [review]
GJ
